### PR TITLE
ci: Switch to GitHub Actions

### PR DESCRIPTION
**Pull request by tavianator**

### CODE REVIEW

**Comment by Mmdixon** [comment]
You should probably update the badges in the readme as well?

File: .github/workflows/ci.yml
Comment (Mmdixon) [inline .github/workflows/ci.yml:3]: The azure pipelines trigger was set for commits/PRs on master only. Did you want to change this behavior?
Otherwise might look like? (push will trigger after merge, but if you want it before merge during PR but not on the commit of the other branch)
```
on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master
```

File: README.rst
Comment (Mmdixon) [inline README.rst:62]: Do you know why the latest version of the redthedocs badge displays as failing?

Comment (tavianator) [reply README.rst:62]: https://github.com/readthedocs/readthedocs-docker-images/issues/107#issuecomment-684021245

I'm gonna try to work around that next :)

File: .github/workflows/ci.yml
Comment (tavianator) [reply .github/workflows/ci.yml:3]: Ah yeah that's more like what I want

File: README.rst
Comment (Mmdixon) [reply README.rst:62]: Ah so the "latest" docs https://bistring.readthedocs.io/en/latest/ is actually an old commit (https://github.com/microsoft/bistring/commit/b79007e9)? But the badge is referring to the actual latest build?

**Review by Mmdixon — APPROVED** [review]
LGTM unless you had something for the readthedocs badge in this PR as well.

File: README.rst
Comment (tavianator) [reply README.rst:62]: Yeah

**Comment by tavianator** [comment]
Nope I'll do that separately

**Pull request merged**
